tests/GenerativeAbstractClassesTests.fs

```diff
@@ -0,0 +1,98 @@
+#if INTERACTIVE
+#load "../src/ProvidedTypes.fsi" "../src/ProvidedTypes.fs" 
+#load "../src/ProvidedTypesTesting.fs"
+
+#else
+
+module FSharp.TypeProviders.SDK.Tests.GenerativeAbstractClassesTests
+#endif
+
+#nowarn "760" // IDisposable needs new
+
+#if !NO_GENERATIVE
+
+open System
+open System.Reflection
+open Microsoft.FSharp.Core.CompilerServices
+open Xunit
+open ProviderImplementation.ProvidedTypes
+open ProviderImplementation.ProvidedTypesTesting
+
+
+[<TypeProvider>]
+type GenerativeAbstractClassesProvider (config: TypeProviderConfig) as this =
+    inherit TypeProviderForNamespaces (config)
+
+    let ns = "AbstractClasses.Provided"
+    let tempAssembly = ProvidedAssembly()
+    let container = ProvidedTypeDefinition(tempAssembly, ns, "Contracts", Some typeof<obj>, isErased = false)
+
+    let createAbstractClass name (members: (string * (string * Type) list * Type * bool) list) =
+        let t = ProvidedTypeDefinition(name, Some typeof<obj>, hideObjectMethods = true, isErased = false)
+        t.SetAttributes(TypeAttributes.Public ||| TypeAttributes.AutoClass ||| TypeAttributes.AnsiClass ||| TypeAttributes.Abstract ||| TypeAttributes.BeforeFieldInit)
+        
+        members
+        |> List.map (fun (name, parameters, retType, isVirtual) ->
+            let ps =
+                parameters
+                |> List.map (fun (name, ty) ->
+                    ProvidedParameter(name, ty))
+            if isVirtual then
+                //let m = ProvidedMethod(name, ps, retType)
+                let m = ProvidedMethod(name, ps, retType, invokeCode = fun args ->
+                    <@ raise (NotImplementedException(name + " is not implemented")) @>.Raw
+                    )
+                m.SetMethodAttrs (MethodAttributes.Public ||| MethodAttributes.HideBySig ||| MethodAttributes.NewSlot ||| MethodAttributes.Abstract ||| MethodAttributes.Virtual)
+                m
+            else
+                let m = ProvidedMethod(name, ps, retType)
+                m.SetMethodAttrs (MethodAttributes.Public ||| MethodAttributes.HideBySig ||| MethodAttributes.NewSlot ||| MethodAttributes.Abstract ||| MethodAttributes.Virtual)
```

Comment:
@panesofglass you add `MethodAttributes.Virtual` attribute for `not isVirtual` case, this is probably the source for 
```
 System.TypeLoadException : Abstract method with non-zero RVA.
```